code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ServiceActivate.java

```diff
@@ -0,0 +1,246 @@
+            }
+        }
+        
+        for (Service consumedService : consumedServices) {
+            try {
+                objectProcessManager.scheduleProcessInstance(ServiceDiscoveryConstants.PROCESS_SERVICE_ACTIVATE,
+                        consumedService, data);
+            } catch (ProcessCancelException e) {
+                // do nothing
+            }
+        }
+        waitForServiceActivate(consumedServices);
+    }
+
+    private void waitForServiceActivate(List<Service> servicesToActivate) {
+        for (Service service : servicesToActivate) {
+            progress.checkPoint("activate consumed service " + service.getName());
+            service = resourceMonitor.waitFor(service, new ResourcePredicate<Service>() {
+                @Override
+                public boolean evaluate(Service obj) {
+                    return CommonStatesConstants.ACTIVE.equals(obj.getState());
+                }
+            });
+        }
+    }
+
+    private void createServiceInstances(Service service, int scale) {
+
+        List<Instance> instancesToStart = new ArrayList<>();
+        for (int i = 0; i < scale; i++) {
+            Instance instance = createInstance(service, i);
+            instancesToStart.add(instance);
+            createInstanceServiceMap(instance, service);
+        }
+        startServiceInstances(instancesToStart);
+    }
+
+    private void startServiceInstances(List<Instance> instancesToStart) {
+        for (Instance instance : instancesToStart) {
+            scheduleStart(instance);
+        }
+        for (Instance instance : instancesToStart) {
+            progress.checkPoint("start service instance " + instance.getUuid());
+            instance = resourceMonitor.waitFor(instance, new ResourcePredicate<Instance>() {
+                @Override
+                public boolean evaluate(Instance obj) {
+                    return InstanceConstants.STATE_RUNNING.equals(obj.getState());
+                }
+            });
+        }
+    }
+
+    protected void scheduleStart(final Instance instance) {
+        if (InstanceConstants.STATE_STOPPED.equals(instance.getState())) {
+            DeferredUtils.nest(new Callable<Object>() {
+                @Override
+                public Object call() throws Exception {
+                    objectProcessManager.scheduleProcessInstance(InstanceConstants.PROCESS_START, instance, null);
+                    return null;
+                }
+            });
+        }
+    }
+
+    private Instance createInstance(Service service, int i) {
+        Environment env = objectManager.findOne(Environment.class, ENVIRONMENT.ID, service.getEnvironmentId());
+        String instanceName = String.format("%s_%s_%d", env.getName(), service.getName(), i + 1);
+        Instance instance = objectManager.findOne(Instance.class, INSTANCE.NAME, instanceName,
+                INSTANCE.REMOVED, null);
+        
+        if (instance == null || instance.getRemoved() != null) {
```

Comment:
Don't need the instance.getRemoved() check since it should be filtered out already